internal/gatewayapi/translator.go

```diff
@@ -406,16 +436,28 @@ func (t *Translator) ProcessListeners(gateways []*GatewayContext, xdsIR *ir.Xds,
 
 			listener.SetCondition(v1beta1.ListenerConditionReady, metav1.ConditionTrue, v1beta1.ListenerReasonReady, "Listener is ready")
 
-			irListener := &ir.HTTPListener{
+			// Add the listener to the Xds IR.
+			xdsIRListener := &ir.HTTPListener{
 				Name:    irListenerName(listener),
 				Address: "0.0.0.0",
 				Port:    uint32(listener.Port),
 				TLS:     irTLSConfig(listener.tlsSecret),
 			}
 			if listener.Hostname != nil {
-				irListener.Hostnames = append(irListener.Hostnames, string(*listener.Hostname))
+				xdsIRListener.Hostnames = append(xdsIRListener.Hostnames, string(*listener.Hostname))
+			}
+			xdsIR.HTTP = append(xdsIR.HTTP, xdsIRListener)
+
+			// Add the listener to the Infra IR proxy listener.
+			infraIRListener := ir.ProxyListener{
+				Ports: []ir.ListenerPort{
+					{
+						Name: string(listener.Name),
+						Port: int32(listener.Port),
```

Comment:
We'll need to sort out how ports on the service map to ports that Envoy listens on. Right now they're 1:1, but I imagine inside the Envoy container we'll want to listen on e.g. 8080/8443, and then have the Service map 80->8080 and 443->8443. However, since we need to handle arbitrary ports, not just those two, we'll need some kind of mapping scheme. I'll file an issue for this.